### PR TITLE
Improved the float and int cli tests and add test for notebook_dir

### PR DESCRIPTION
**Pull request by aaronmak**

### CODE REVIEW

File: papermill/tests/test_cli.py [hunk 1]
```diff
@@ -23,10 +23,12 @@ def test_resolve_type(test_input, expected):
 
 @pytest.mark.parametrize("value,expected", [
     (13.71, True),
+    ('12.51', True),
     ("False", False),
     ("None", False),
     (-8.2, True),
-    (10, False),
+    (10, True),
+    ('10', True),
```
Comment (rgbkrk) [inline papermill/tests/test_cli.py]: Excellent!

File: papermill/cli.py [hunk 2]
```diff
@@ -88,26 +88,39 @@ def _resolve_type(value):
...
-    if type(value) is int:
-        return True
-    else:
+    """
+    Casting is used to check if the value
+    in the cli can be converted to an int
+    """
+    try:
+        float_cast = float(value)
+        int_cast = int(float_cast)
```
Comment (rgbkrk) [inline papermill/cli.py]: Instead of doing this here, I'd keep the ordering of where `_is_float` is in the `_resolve_type` check.

**Comment by rgbkrk** [comment]
I highly recommend creating branches from `master`, following [github flow](https://guides.github.com/introduction/flow/) when making PRs. This PR has more commits than it should because you're building off your master branch which is out of sync with the "upstream" master. Since there are conflicts, you need to rebase your commits, which is more hassle than necessary if building from `nteract/papermill`'s `master`.

To make this simple, I'd see if you can fetch from this repository and create a new branch from here, then push that up as a new 

```
# Within the papermill repo, at the root
git checkout -b testbranch
git remote add upstream https://github.com/nteract/papermill
git fetch upstream
git reset --hard upstream/master

# Now apply changes
```

That should hopefully make a nice clean branch you can make the changes to.

**Comment by aaronmak** [comment]
that's a good point, I'll create a new branch and work from there

**Pull request closed without merging**